plugin/storage/es/spanstore/reader_test.go

```diff
@@ -93,11 +94,19 @@ func withSpanReader(fn func(r *spanReaderTest)) {
 		client:    client,
 		logger:    logger,
 		logBuffer: logBuffer,
-		reader:    NewSpanReader(client, logger),
+		reader:    newSpanReader(client, logger),
 	}
 	fn(r)
 }
 
+func TestNewSpanReaderWithMetrics(t *testing.T) {
+	client := &mocks.Client{}
+	logger, _ := testutils.NewLogger() // logBuffer unneeded
+	metricsFactory := metrics.NewLocalFactory(0)
+	var reader spanstore.Reader = NewSpanReader(client, logger, metricsFactory)
```

Comment:
technically not needed, but works as a check so that the spanReader implements spanstore.Reader